src/library/dao/trackdao.cpp

```diff
@@ -702,6 +702,21 @@ TrackPointer TrackDAO::addTracksAddFile(const QFileInfo& fileInfo, bool unremove
         qDebug() << "TrackDAO::addTracksAddFile:"
                 << "Track has already been added to the database"
                 << oldTrackId;
+        DEBUG_ASSERT(pTrack->getDateAdded().isValid());
+        const auto trackLocation = pTrack->getLocation();
+        // TODO: These duplicates are only detected by chance when
+        // the other track is currently cached. Instead file aliasing
+        // must be detected reliably in any situation.
+        if (fileInfo.absoluteFilePath() != trackLocation) {
+            kLogger.warning()
+                    << "Both track locations"
```

Comment:
How about something like this: 
<< "Cannot add Track: "
<< fileInfo.absoluteFilePath()
<< because it is linked to: "
<< fileInfo.canonicalFilePath();
<< which is already in the library 
<< as : 
<< trackLocation

Reply:
It is non-deterministic which file is picked up first by the library scanner, either a symbolic link or the actual file. Checking all possible cases and adding a separate message for each of them is not worth the effort. I will not do this.

Reply:
Your proposed message is wrong when scanning the linked example folder. In my tests, the symbolic link to the file was added to the database first and the actual file was rejected.

Reply:
The issue I am trying to solve is to give the users some hints what happens, that's all. 
"Both track locations ..." does not fit to this, because the user has only requested one track.

Reply:
Thank you.